### DISCUSSION TITLE
Video Streaming with Yolov8

### DISCUSSION THREAD

**Discussion by jcgds** [ultralytics/ultralytics#5258] [Q&A]
Hello, I'm using Yolov8 with Roboflow Supervision. In my project, I need to create an RTSP video stream from my client's on-premises environment to perform processing in Microsoft Azure. Is there any ultralytics/yolo reference architecture for this type of project? What is the best way to carry out this streaming with YOLOv8 and supervise it? Thanks!

**Comment by glenn-jocher** [comment]
@jcgds hello!

For setting up an RTSP video stream with YOLOv8, you would typically capture the stream using a suitable library in your environment, process the frames with YOLOv8 for detection, and then send the processed frames to your desired destination. While we don't have a specific reference architecture for streaming with YOLOv8, the process generally involves the following steps:

1. Capture the RTSP stream using a library that can handle video streaming protocols.
2. Process the frames with YOLOv8's Predict mode to perform object detection.
3. Supervise and monitor the detections as needed.
4. Optionally, re-stream the processed video to Azure or store the results.

For supervision, you can use the Track mode if you need object tracking capabilities. Remember to check out the Predict and Track sections in our documentation for more details on how to use these modes effectively.

If you need further guidance on setting up your specific architecture, you might consider exploring the Enterprise License, which offers more flexibility and support for commercial applications.

Good luck with your project! 🚀
